plugins/inputs/socket_listener/socket_listener.go

```diff
@@ -112,6 +117,7 @@ type SocketListener struct {
 	ServiceAddress string
 	MaxConnections int
 	ReadBufferSize int
+	ReadTimeout    int
```

Comment:
Sorry, I should have been more clear, this is the value that should be an `internal.Duration`.  This will allow people to specify the timeout like "30s", so make sure to update the README as well.